# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.h
# Commit: 3cfa16c9f9a9
# Full Hash: 3cfa16c9f9a97c5963b31a9ca080df29e772836c
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-30 16:09:44
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 3: Implement the algorithm for obtaining tls and use it for wasm signal handling. r=lth
#   
#   
#   This is the third part of series of patches to Frame without tls pointer.
#   Here we preserve initial tls in all entry stubs and then use it to find a proper tls instance for a given frame.
# ==============================================================================

diff -r 50772b4ff6aa -r 3cfa16c9f9a9 js/src/wasm/WasmFrameIter.h
--- a/js/src/wasm/WasmFrameIter.h	Fri Oct 23 18:17:37 2020 +0000
+++ b/js/src/wasm/WasmFrameIter.h	Fri Oct 23 18:17:37 2020 +0000
@@ -233,6 +233,10 @@
 void GenerateFunctionEpilogue(jit::MacroAssembler& masm, unsigned framePushed,
                               FuncOffsets* offsets);
 
+// Iterates through frames for either possible cross-instance call or an entry
+// stub to obtain tls that corresponds to the passed fp.
+TlsData* GetNearestEffectiveTls(Frame* fp);
+
 // Describes register state and associated code at a given call frame.
 
 struct UnwindState {